src/java.base/share/classes/java/text/DigitList.java

```diff
@@ -788,23 +788,23 @@ public String toString() {
         if (isZero()) {
             return "0";
         }
-        StringBuffer buf = getStringBuffer();
-        buf.append("0.");
-        buf.append(digits, 0, count);
-        buf.append("x10^");
-        buf.append(decimalAt);
-        return buf.toString();
+        StringBuilder temp = getStringBuilder();
+        temp.append("0.");
+        temp.append(digits, 0, count);
+        temp.append("x10^");
+        temp.append(decimalAt);
+        return temp.toString();
     }
 
-    private StringBuffer tempBuffer;
+    private StringBuilder tempBuilder;
 
-    private StringBuffer getStringBuffer() {
-        if (tempBuffer == null) {
-            tempBuffer = new StringBuffer(MAX_COUNT);
+    private StringBuilder getStringBuilder() {
+        if (tempBuilder == null) {
+            tempBuilder = new StringBuilder(MAX_COUNT);
         } else {
-            tempBuffer.setLength(0);
+            tempBuilder.setLength(0);
         }
-        return tempBuffer;
+        return tempBuilder;
     }
```

Comment:
Is it worth considering whether it's still necessary to cache and reuse a single String[Buffer|Builder] instance when using 2022 garbage collectors?